types/context.go

```diff
@@ -277,21 +277,24 @@ func (c Context) Value(key interface{}) interface{} {
 // ----------------------------------------------------------------------------
 
 // KVStore fetches a KVStore from the MultiStore.
-func (c Context) KVStore(key storetypes.StoreKey) KVStore {
-	return gaskv.NewStore(c.MultiStore().GetKVStore(key), c.GasMeter(), c.kvGasConfig)
+// NOTE: Uses pointer receiver to save on execution time.
+func (c *Context) KVStore(key storetypes.StoreKey) KVStore {
+	kv := c.ms.GetKVStore(key)
+	return gaskv.NewStore(kv, c.gasMeter, c.kvGasConfig)
 }
```

Comment:
Callers who invoke c.KVStore will get a KVStore value with a kvGasConfig that is no longer a copy of the config data, but instead is a pointer to the same storetypes.GasConfig value of the Context from which the KVStore was derived. This is a fundamental change to the semantics of the KVStore method, and of the Context API in general. It's important to establish the safety of large changes like this with comprehensive test coverage.

Reply:
Ref: https://github.com/cosmos/cosmos-sdk/pull/14203#discussion_r1042905158

Reply:
+1 if we need testing for this but old behaviour is retained